### PR TITLE
Update and harden systemd service files

### PR DESCRIPTION
**Pull request by djmoch**
- Delete the redundant worker service file

- Adjust the remaining service files to comply with the Systemd file
  system hierarchy [1], which mandates that user-callable binaries live
  in /usr/bin, and that system data live in /var/lib

- Convert service files into template units based on the subdirectory of
  /var/lib where the configuration lives. For instance a master whose
  configuration lives in /var/lib/buildbot/master would be invoked by:

    systemctl start buildbot@master.service

  Likewise, a worker is invoked by:

    systemctl start buildbot-worker@worker.service

- Harden the services by removing their access to /home, giving them
  private /tmp and /dev trees, and effectively removing write access to
  the rest of the system

[1] - https://www.freedesktop.org/software/systemd/man/file-hierarchy.html

### CODE REVIEW

**Comment by djmoch** [comment]
Based on a discussion with @yan12125, who maintains the Buildbot package for Arch Linux.

**Comment by yan12125** [comment]
Thank you very much for the pull request! I'm looking for improved systemd service files upstreamed for inclusion in official Arch Linux packages!

File: master/contrib/systemd/buildbot@.service [hunk 1]
```diff
@@ -1,17 +1,18 @@
...
 After=network.target
 
-
 [Service]
 User=buildbot
 Group=buildbot
-WorkingDirectory=/home/buildbot/master
-ExecStart=/usr/local/bin/buildbot start --nodaemon
+WorkingDirectory=/var/lib/buildbot
+ExecStart=/usr/bin/buildbot --verbose start --nodaemon %I
```
Comment (tardyp) [inline master/contrib/systemd/buildbot@.service:15]: Can you please document that you will need to create the master (and worker) in a subdirectory named after the systemd instanceID?

Comment (djmoch) [reply master/contrib/systemd/buildbot@.service:15]: I've added some commentary to the top of the files that I hope clarifies things. Let me know if there's anything else I can do.

**Comment by yan12125** [comment]
Hi @tardyp, mind to have a look again?

**Pull request merged**